src/Simplex/Chat/Messages.hs

```diff
@@ -715,16 +721,69 @@ instance StrEncoding ACIStatus where
           "snd_rcvd" -> ACIStatus SMDSnd <$> (CISSndRcvd <$> (A.space *> strP) <*> ((A.space *> strP) <|> pure SSPComplete))
           "snd_error_auth" -> pure $ ACIStatus SMDSnd CISSndErrorAuth
           "snd_error" -> ACIStatus SMDSnd . CISSndError . T.unpack . safeDecodeUtf8 <$> (A.space *> A.takeByteString)
+          "snd_delivery_error" -> ACIStatus SMDSnd . CISSndDeliveryError <$> (A.space *> strP)
+          "snd_delivery_warning" -> ACIStatus SMDSnd . CISSndDeliveryWarning <$> (A.space *> strP)
           "rcv_new" -> pure $ ACIStatus SMDRcv CISRcvNew
           "rcv_read" -> pure $ ACIStatus SMDRcv CISRcvRead
           _ -> fail "bad status"
 
+data SndDeliveryError
+  = SDEAuth
+  | SDEQuota
+  | SDEOther String
+  deriving (Eq, Show)
+
+instance StrEncoding SndDeliveryError where
+  strEncode = \case
+    SDEAuth -> "sde_auth"
+    SDEQuota -> "sde_quota"
+    SDEOther e -> "sde_other " <> encodeUtf8 (T.pack e)
+  strP =
+    A.takeWhile1 (/= ' ') >>= \case
+      "sde_auth" -> pure SDEAuth
+      "sde_quota" -> pure SDEQuota
+      "snd_other" -> SDEOther . T.unpack . safeDecodeUtf8 <$> (A.space *> A.takeByteString)
+      _ -> fail "bad SndDeliveryError"
+
+-- see MWARN, serverHostError in agent
+data SndDeliveryWarning
+  = SDWBrokerHost
+  | SDWBrokerTransportVersion
+  | SDWProxyHost
+  | SDWProxyTransportVersion
+  | SDWProxyBrokerHost
+  | SDWProxyBrokerTransportVersion
```

Comment:
```suggestion
  = SndRelayError SrvError -- from BROKER XXX and PROXY BROKER XXX errors
  | SndProxyError SrvError -- from PROXY XXX errors
```

```
data SrvError = SrvHostError | SrvVersionError | SrvError String
```

Reply:
what `SrvError String` should be used for? Currently this type is for errors as differentiated in serverHostError, with SDWOther used for "other" errors. In new version it's unclear how other errors should be translated - as SndRelayError or as SndProxyError.

```
data SndDeliveryWarning =
  = SndRelayError SrvError
  | SndProxyError SrvError
  | SndOtherError String

data SrvError = SrvHostError | SrvVersionError
```

^ Maybe this is how it was intended? Also unclear what's the need for an additional SrvError type, flat list is simpler to maintain and adapt to the needs of agent

Reply:
SndRelayError SrvError -- from BROKER XXX
SndProxyRelayError -- and PROXY BROKER XXX errors